### PR TITLE
refactor: deprecate non uri connection

### LINKED ISSUE #979: provider/postgis: Clean out deprecated connection params
In v0.15.0 support for a connection string uri was added to provider/postgis. When this change was made, we marked the [old connection params as deprecated](https://github.com/go-spatial/tegola/tree/master/provider/postgis#deprecated-connection-properties). These params now need to be removed. This issue is for tracking this removal from the following places:

- [ ] `provider/postgis`:  This package needs to remove support for the old connection params. The README should also be updated.
- [ ] `.github/`: The CI uses the old connection params in a few places and needs to be migrated.

### PR DESCRIPTION
**Pull request by iwpnd**
closes #979

### CODE REVIEW

File: mvtprovider/postgis/README.md [hunk 1]
```diff
@@ -1,32 +1,22 @@
...
 ### Connection Properties
 
-- `name` (string): [Required] provider name is referenced from map layers.
-- `type` (string): [Required] the type of data provider. must be "postgis" to use this data provider
-- `host` (string): [Required] PostGIS database host
-- `port` (int): [Required] PostGIS database port (required)
-- `database` (string): [Required] PostGIS database name
-- `user` (string): [Required] PostGIS database user
-- `password` (string): [Required] PostGIS database password
-- `max_connections` (int): [Optional] The max connections to maintain in the connection pool. Defaults to 100. 0 means no max.
```
Comment (ARolek) [inline mvtprovider/postgis/README.md:28]: We should add a reference around some of these pgx specific configs. It looks like the [`ParseConfig` ](https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool#ParseConfig) function for pgxpool supports the following additional query strings:

* pool_max_conns: integer greater than 0 
* pool_min_conns: integer 0 or greater 
* pool_max_conn_lifetime: duration string 
* pool_max_conn_idle_time: duration string 
* pool_health_check_period: duration string 
* pool_max_conn_lifetime_jitter: duration string

I think adding some notes to the README about how this package uses `pgx` under the hood with a link with to the ParseConfig function and indicating some of the available connection config query strings would be really helpful.

**Review by ARolek — CHANGES_REQUESTED** [review]
Looks really good in here! I only have some minor comments inline, mainly around adding some documentation referencing the pgxpool.ParseConfig function which has additional connection string params that we have deprecated config support for, but are still supported in the connection string.

**Review by gdey — APPROVED** [review]
LGTM, when you have a chance can you rebase the commits into a single narrative.

**Review by ARolek — APPROVED** [review]
Looks clean! Thanks for tackling that. Please squash the commits into a single one and then we can land this PR and get v0.20.0 of tegola released!

**Comment by iwpnd** [comment]
lets go @ARolek

**Pull request merged**